src/app/ui/user-profile/user-profile.component.ts

```diff
@@ -43,12 +47,15 @@ export class UserProfileComponent implements OnInit {
       switchMap((params) => {
         const playerId = params['uid'] as string;
         if (playerId) {
+          this.getTotalExp(playerId);
```

Comment:
please don't do this, lets embrace observables and follow how data is being formed in `getPlayerInfo()`, we should add XP in User's interface.